### PR TITLE
Faction and personal scores 

### PR DESCRIPTION
**Pull request by Evankhell561**
## About The Pull Request

adds the faction and personal scores and displays them in the end round.

## Why It's Good For The Game
kot map

## Changelog
:cl:
add: faction and personal scores
/:cl:

### CODE REVIEW

File: code/modules/research/research.dm [hunk 1]
```diff
@@ -29,6 +29,11 @@ Procs:
 **	Includes all the helper procs and basic tech processing.  **
 ***************************************************************/
 
+/datum/research/proc/adjust_research_points(varlue)
+	if(varlue > 0)
+		GLOB.research_point_gained += varlue
```
Comment (TorinoFermic) [inline code/modules/research/research.dm:34]:
```suggestion
	if(value > 0)
		GLOB.research_point_gained += value
```

File: code/game/gamemodes/scores.dm [hunk 2]
```diff
@@ -0,0 +1,393 @@
+#define MAX_FACTION_SCORE 1000
```
Comment (TheShown911) [inline code/game/gamemodes/scores.dm:1]: Faction score should be uncapped if it's just a number so we can easily spot exploits(like the guild infinite money with surplus ammo), but if it's going to get tied to some other important system (like irska coins) it should be capped.

File: code/game/gamemodes/scores.dm [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+	"}
+
+	if(is_scored_departmen())
+		final_score += get_faction_score()
+		max_personal_score += MAX_FACTION_SCORE
+
+	dat += "<br><hr>"
+	dat += "<b><u>Your total score is:</u></b> [get_color_score(final_score, final_score, max_personal_score)] Points<br>"
+
+	src << browse(dat, "window=roundstats;size=500x600")
```
Comment (TheShown911) [inline code/game/gamemodes/scores.dm:363]: Would like a picture of how this looks in game.

**Review by TheShown911 — APPROVED** [review]
Overall very excited for this PR, would like it if players could see what they need to do to have their faction score high somewhere in the job selection screen, but this can be merged without it.

File: code/game/gamemodes/scores.dm (hunk 2 above)
Comment (Evankhell561) [reply code/game/gamemodes/scores.dm:1]: This is not a cap, it is simply a maximum reference value. If the score is less than half this value, it means that a good job has not been done in the round.

File: code/datums/objective/individual_objective/individual_objective.dm [hunk 4]
```diff
@@ -27,16 +30,16 @@
...
-	var/allow_cruciform = TRUE
 	var/units_completed = 0
 	var/units_requested = 1
-	var/based_time = FALSE
+	var/completed = FALSE
 	var/list/req_department = list()
 	var/req_cruciform = FALSE
-	var/insight_reward = 20
+	var/allow_cruciform = TRUE
+	var/based_time = FALSE
```
[the remark this answers is not in the document]
Comment (Evankhell561) [reply code/datums/objective/individual_objective/individual_objective.dm:39]: > Based time.

?

[the remark this answers is not in the document]
Comment (TheShown911) [reply code/datums/objective/individual_objective/individual_objective.dm:39]: Based time.

[the remark this answers is not in the document]
Comment (Evankhell561) [reply code/datums/objective/individual_objective/individual_objective.dm:39]: > Based time.

?

[the remark this answers is not in the document]
Comment (TheShown911) [reply code/datums/objective/individual_objective/individual_objective.dm:39]: You poor soul, there are no problems with the code i just find based time funny

**Comment by Evankhell561** [comment]
:pray:

**Comment by Evankhell561** [comment]
image
image
image

**Comment by Evankhell561** [comment]
fixed typos:
image

**Pull request merged**